### ISSUE TITLE
Add information about the repository, to which a script belongs

### ISSUE THREAD

**Issue by madoar** [PlayOnLinux/POL-POM-5#656]
Currently a user sees no information about the repository to which a script belongs.
This is currently no big problem, because each script exists only once. 
But this may change, if we have two repositories in the future, one with a stable version of a script, and another with a development version of a script, it is possible that two, or more scripts for the same application, with the same name, exist. 
If such a situation occurs, POL 5 needs to show the source repository of the script. I think the best location to show the source repository is both in the detailed view of the script and directly in the "Apps" tab for each visible script.

**Comment by plata** [comment]
You can add this to the ApplicationDTO. A simple string should be enough (e.g. "git: https://...).

**Comment by madoar** [comment]
You mean as a new variable inside the ApplicationDTO class?
How should this variable be shown best to the user?

**Comment by plata** [comment]
Yes. In the AppPanel. Something like "Source: git: https://...".

**Comment by madoar** [comment]
When implementing this, should we also add a field for the last modification of the script at the same time? As far as I remember we found no better solution that to add a modification date field to the script in https://github.com/PlayOnLinux/Scripts/issues/188.

**Comment by plata** [comment]
I wouldn't do it in the same PR (you have to touch all scripts).

**Comment by madoar** [comment]
This seems to be more work than I initially thought. The reason for this is, that `GitApplicationsSource` internally uses `LocalApplicationsSource` to process the folder after the repository has been locally cloned. 
The problem here is, that `LocalApplicationsSource` doesn't know, and truthfully spoken shouldn't care in my eyes, if the folder, it is processing, is a git repository or not. Therefore it can't set the git repository url as the source. The only thing it can set as the source is the folder name, which we don't want in case of a git repository. 

I would propose, that we move the folder-processing-parts from `LocalApplicationsSource` to a new class/interface. Then we can easily extend `LocalApplicationsSource` and `GitApplicationsSource` from the new class/interface, to mixin the ability to read the corresponding folder. After the categories have been fetched, we set the source information of the `ApplicationDTO`s, which is still easily doable because this is still done in the ApplicationSource `fetchInstallableApplications` method.  
What do you think?

**Comment by plata** [comment]
Can't you just set the git repository URL after
```Java
this.cache = localAppsSourceFactory.createInstance(gitRepositoryLocation.getAbsolutePath())
					.fetchInstallableApplications();
```

**Comment by madoar** [comment]
Yes I could do it after setting the cache, but then I would simply override the source information, that the LocalApplicationsSource set. 
In addition it could be a possibility that LocalApplicationsSource sets some other values, which are wrong or not fitting for a GitApplicationsSource, if not now then maybe at a later time. To prevent such problems I would like to move the parts that access the filesystem and parses the scripts and categories to another abstract class or interface.

**Comment by plata** [comment]
I wouldn't have a problem with the simple solution. But if you prefer to have it cleaner in a new class, that's also fine for me.

**Comment by madoar** [comment]
I've just discovered, that an `ApplicationDTO` can have multiple `ScriptDTO`s.
While taking a more detailed look at the code, it seems possible, that these multiple `ScriptDTO`s can come from multiple `ApplicationsSource`s, through their merging (see class/interface MergeableApplicationsSource in PR #657).
Therefore it seems better to add a source variable to the `ScriptDTO` class instead of the `ApplicationDTO`.
What do you think?

**Comment by qparis** [comment]
I really think that we should keep things really simple.

Where do you want this info to be displayed?

**Comment by madoar** [comment]
I thought about adding it behind the script name inside the AppPanel behind the script name (see `AppPanel.java` line `70`).
